### PR TITLE
CDAP-14827 Added admin call to get namespace information

### PR DESCRIPTION
**Pull request by albertshau**
Added a method to get information about a namespace through the
Admin. Also added a generation field to namespaces that can
be used by consumers to know when a namespace has been deleted
and then recreated.

This feature will be used in dataprep to include the namespace
generation in entity row keys to avoid showing entities from
a previously existing namespace that was deleted and then
recreated.

### CODE REVIEW

**Comment by albertshau** [comment]
https://builds.cask.co/browse/CDAP-DUT6763

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/DefaultAdmin.java [hunk 1]
```diff
@@ -174,4 +177,19 @@ public boolean namespaceExists(String namespace) throws IOException {
       throw new IOException(e);
     }
   }
+
+  @Nullable
+  @Override
+  public NamespaceSummary getNamespaceSummary(String namespace) throws IOException {
+    try {
+      NamespaceMeta meta = namespaceQueryAdmin.get(new NamespaceId(namespace));
+      return new NamespaceSummary(meta.getName(), meta.getDescription(), 0L);
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/DefaultAdmin.java]: Why the generation is always 0?

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/AppFabricTestHelper.java [hunk 2]
```diff
@@ -194,7 +194,9 @@ private static void ensureNamespaceExists(NamespaceId namespaceId, CConfiguratio
     NamespaceAdmin namespaceAdmin = getInjector(cConf).getInstance(NamespaceAdmin.class);
     try {
       if (!namespaceAdmin.exists(namespaceId)) {
-        namespaceAdmin.create(new NamespaceMeta.Builder().setName(namespaceId).build());
+        namespaceAdmin.create(new NamespaceMeta.Builder()
```
Comment (chtyim) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/AppFabricTestHelper.java]: Seems like an unnecessary change?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/DefaultAdmin.java (hunk 1 above)
Comment (albertshau) [reply cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/DefaultAdmin.java]: ugh sorry, this is incomplete.

**Review by chtyim — APPROVED** [review]
LGTM
